agent/engine/docker_task_engine.go

```diff
@@ -641,6 +641,29 @@ func (engine *DockerTaskEngine) pullContainer(task *api.Task, container *api.Con
 		if emptyvolume.LocalImage {
 			return engine.client.ImportLocalEmptyVolumeImage()
 		}
+	default:
+		// If the agent pull behavior is never, we don't try to pull the image,
+		// try to use local image cache instead.
+		if engine.cfg.AgentPullBehavior == config.NeverAgentPullBehavior {
```

Comment:
can you please refactor these into a method? May be something that returns a boolean to indicate if you should return `dockerapi.DockerContainerMetadata{Error: nil}`?

Reply:
OK, will abstract it out.